core/src/main/java/org/jdbi/v3/core/statement/SqlStatement.java

```diff
@@ -1250,20 +1263,34 @@ public final This bindBySqlType(int position, Object value, int sqlType) {
      *     .list();
      * </pre>
      *
-     * @param key    attribute name
-     * @param values vararg values that will be comma-spliced into the defined attribute value.
+     * @param key     attribute name
+     * @param onEmpty handler for null/empty value array
+     * @param values  vararg values that will be comma-spliced into the defined attribute value.
      * @return this
      * @throws IllegalArgumentException if the vararg array is empty.
+     * @see BindListEmptyHandling
      */
-    public final This bindList(String key, Object... values) {
-        if (values.length == 0) {
-            throw new IllegalArgumentException(
-                    getClass().getSimpleName() + ".bindList was called with no vararg values.");
+    public final This bindList(String key, BiConsumer<SqlStatement, String> onEmpty, Object... values) {
```

Comment:
the odds of someone legit passing a `BiConsumer<SqlStatement, String>` as a vararg value to be bound as a query param, which would cause this method to falsely match and break their query, are ridiculous...

Reply:
Would it be better to move `onEmpty` to the first argument?  That both avoids this caveat (even though I agree it'd be a ridiculous situation) and it also avoids splitting `key` from `values` which are the "most related" arguments.

Reply:
Hmm why not